### DISCUSSION TITLE
Rate limiting endpoints

### DISCUSSION THREAD

**Discussion by pratapvardhan** [gramener/gramex#384] [Ideas]
Usecase: APIs rate-limited by user roles. How should gramex enable API rate-limiting?

- Based on role usage limit
- Throttle load for burst usage
- Should it be configuration enabled or handled within functions?

Would be super useful for building APIs and SaaS products.

**Comment by sanand0** [comment]
Agreed.

Throttling for burst usage is independent of role-based usage, and I think we can handle this via caching. So I'd ignore this need for now

I suggest having this data-based rather than configuration-based or function-based. We could have a table like this:

![image](https://user-images.githubusercontent.com/63734/110803598-5c9deb00-82a5-11eb-86f7-36a5e8e25b23.png)

... and allow admins to edit such a table.
